br/pkg/lightning/errormanager/errormanager.go

```diff
@@ -264,20 +274,23 @@ func (em *ErrorManager) RecordIndexConflictError(
 	if em.db == nil {
 		return nil
 	}
+	if len(conflictInfos) == 0 {
+		return nil
+	}
 
 	exec := common.SQLWithRetry{
 		DB:           em.db,
 		Logger:       logger,
 		HideQueryLog: redact.NeedRedact(),
 	}
 	return exec.Transact(ctx, "insert index conflict error record", func(c context.Context, txn *sql.Tx) error {
-		stmt, err := txn.PrepareContext(c, fmt.Sprintf(insertIntoConflictErrorIndex, em.schemaEscaped))
-		if err != nil {
-			return err
-		}
-		defer stmt.Close()
+		buf := &bytes.Buffer{}
```

Comment:
use `string.Builder` should be a more ergomonic way here, though there is no much difference.

Reply:
They are so similar 😂. I will replace it with `strings.Builder`.